### ISSUE TITLE
[doc] Update wiki for catppuccin?

### ISSUE THREAD

**Issue by Jint-lzxy** [ayamir/nvimdots#185]
As mentioned in #163. I don't have write access to the wiki... so here's the issue instead 🥳 You can review these changes and pick the appropriate lines that could be added to the wiki.

~**NOTE: There's a new PR (#183) modifying some parts of the catppuccin configs, which means some links in the wiki need to be changed. There would be some changes when #183 is merged.**~ Now replaced.

## What's inside?
- Introduction
- How-to-use
  - Basic usage:
    - Customizing configs / integrations.
    - Enable the **auto-compile** feature _(default)_
  - Advanced usage:
    - Customizing palettes _(per flavour)_.
    - Customizing highlight.
    - Customizing **auto-compile** hook.

## Preview
<details>
    <summary>Raw markdown code <i>(add this to the wiki!)</i></summary>
<p>

`````````markdown
# Guide on how to use the catppuccin colorscheme
## What is Catppuccin? <sup><a href="https://www.reddit.com/r/neovim/comments/r9619t/comment/hna3hz8">[1]</a></sup>
> Catppuccin is a community-driven soothing pastel theme that aims to be the middle ground between low and high-contrast themes, providing a warm color palette with 26 eye-candy colors that are bright enough to be visible during the day, yet pale enough to be easy on your eyes throughout the night.

## Basic Usage
Modify [these lines](https://github.com/ayamir/nvimdots/blob/94f153471068c5145b2036be1f688c0f954cd973/lua/modules/ui/config.lua#L191-L267). _(Note: This link might be slightly different from `HEAD`, but it can be used as a reference.)_ See detailed explanation of each option below.

### General
These settings are unrelated to any group and are globally independent.
-   `vim.g.catppuccin_flavour`: _(Can be any one of: `latte`, `frappe`, `macchiato`, or `mocha`)_ this is **mandatory**. You **must** set this value in order to make catppuccin work correctly. Note that `latte` is a light colorscheme, and the rest are dark schemes; The `mocha` palette is the only one that has been modified to make catppuccin look like the v0.1 one. Check out [this PR](https://github.com/ayamir/nvimdots/pull/163) for details.
-   `transparent_background`: _(Boolean)_ if true, disables setting the background color.
-   `term_colors`: _(Boolean)_ if true, sets terminal colors (a.k.a., `g:terminal_color_0`).

### Dim inactive
This setting manages the ability to dim **inactive** splits/windows/buffers.
-   `enabled`: _(Boolean)_ if true, dims the background color of inactive window or buffer or split.
-   `shade`: _(string)_ sets the shade to apply to the inactive split or window or buffer.
-   `percentage`: _(number from 0 to 1)_ percentage of the shade to apply to the inactive window, split or buffer.

### Styles
Handles the style of general highlight groups _(see `:h highlight-args` for detailed explanation)_:
-   `comments`: _(Table)_ changes the style of comments.
-   `functions`: _(Table)_ changes the style of functions  _(e.g., [`set_auto_compile`](https://github.com/ayamir/nvimdots/blob/94f153471068c5145b2036be1f688c0f954cd973/lua/modules/ui/config.lua#L188) in config)_.
-   `keywords`: _(Table)_ changes the style of keywords _(e.g., `local`)_.
-   `strings`: _(Table)_ changes the style of strings.
-   `variables`: _(Table)_ changes the style of variables.
-   `properties`: _(Table)_ changes the style of a phantom field with only getter and/or setter _(e.g., field access `tbl.field`)_.
-   `operators`: _(Table)_ changes the style of operators.
-   `conditionals`: _(Table)_ changes the style of conditional check keywords _(e.g., `if`)_.
-   `loops`: _(Table)_ changes the style of loop keywords _(e.g., `for`)_.
-   `booleans`: _(Table)_ changes the style of booleans.
-   `numbers`: _(Table)_ changes the style of numbers.
-   `types`: _(Table)_ changes the style of types _(e.g., `int`)_.

### Integrations
These integrations allow catppuccin to set the theme of various plugins. To enable an integration you need to set it to `true`.

### Using the auto-compile feature
> Catppuccin is a highly customizable and configurable colorscheme. This does however come at the cost of complexity and execution time.

Catppuccin can pre-compute the results of configuration and store the results in a compiled lua file. These pre-cached values are later used to set highlights. The cached file is stored at `vim.fn.stdpath("cache") .. "/catppuccin"` by default _(use `:lua print(vim.fn.stdpath("cache") .. "/catppuccin")` to see where it locates on your computer)_. You may change this behavior by modifying [this line](https://github.com/ayamir/nvimdots/blob/94f153471068c5145b2036be1f688c0f954cd973/lua/modules/ui/config.lua#L202).

If you would like to disable this feature, set [`enable_compile`](https://github.com/ayamir/nvimdots/blob/94f153471068c5145b2036be1f688c0f954cd973/lua/modules/ui/config.lua#L187) to false. However, this is **NOT** recommended because it could result in a slower startup time.

There's also an auto-compile hook that could compile catppuccin every time packer compiles _(i.e., the execution of `:PackerCompile`)_. Disable this behavior by commenting out [this line](https://github.com/ayamir/nvimdots/blob/94f153471068c5145b2036be1f688c0f954cd973/lua/modules/ui/config.lua#L188). _(You may also want to see its implementation - that's right above this line)_. However, after disabling this feature, you'll have to **manually** compile catppuccin using `:CatppuccinCompile` every time you modify its settings! _(If you enable catppuccin's compile feature, of course)._ Otherwise, the changes will **NOT** have any effect.

## Advanced Feature
### Customizing the palette
Not satisfied with the current appearance? You may modify the palette yourself, like `mocha`!

#### Get catppuccin colors
```lua
local latte = require("catppuccin.palettes").get_palette "latte"
local frappe = require("catppuccin.palettes").get_palette "frappe"
local macchiato = require("catppuccin.palettes").get_palette "frappe"
local mocha = require("catppuccin.palettes").get_palette "mocha"
vim.g.catppuccin_flavour = "macchiato" -- Has to be set in order for empty argument to work
local colors = require("catppuccin.palettes").get_palette() -- g:catppuccin_flavour's palette
```

These lines would all return a table respectively, where the key is the name of the color and the value is its hex value.

#### Overwriting highlight groups
**Important Note:** If you also modified the color for any of the palettes at the same time, due to the fact that the changes made to the palette(s) when we're overwriting the highlight group have not been actually overwritten, and in the case of calling setup only once, the color we passed in has already been parsed (for example, `cp.pink -> "#F5C2E7"`), so it is **completely impossible** to obtain the changed palette, which means you need to manually set the new color after obtaining the palette, [like so](https://github.com/ayamir/nvimdots/blob/94f153471068c5145b2036be1f688c0f954cd973/lua/modules/ui/config.lua#L126-L164).

Global highlight groups can be overwritten starting from [this line](https://github.com/ayamir/nvimdots/blob/94f153471068c5145b2036be1f688c0f954cd973/lua/modules/ui/config.lua#L299) like so:

```lua
custom_highlights = {
	<hi_group> = { <fields> }
}
```

Here is an example:

```lua
vim.g.catppuccin_flavour = "macchiato"
local colors = require("catppuccin.palettes").get_palette() -- fetch colors from g:catppuccin_flavour palette
require("catppuccin").setup {
	custom_highlights = {
		Comment = { fg = colors.flamingo },
		TSConstBuiltin = { fg = colors.peach, style = {} },
		TSConstant = { fg = colors.sky },
		TSComment = { fg = colors.surface2, style = { "italic" } }
	}
}
```

Per flavour highlight groups can be overwritten like so:

```lua
highlight_overrides = {
	all = { -- Will be replaced with custom_highlights if it exists
		<hi_group> = { <fields> }
	}, -- Same for each flavour
	latte = {},
	frappe = {},
	macchiato = {},
	mocha = {},
}
```

Here is an example:

```lua
local ucolors = require "catppuccin.utils.colors"
local latte = require("catppuccin.palettes").get_palette "latte"
local frappe = require("catppuccin.palettes").get_palette "frappe"
local macchiato = require("catppuccin.palettes").get_palette "macchiato"
local mocha = require("catppuccin.palettes").get_palette "mocha"
vim.g.catppuccin_flavour = "macchiato"
local colors = require("catppuccin.palettes").get_palette() -- return vim.g.catppuccin_flavour palette
require("catppuccin").setup {
	highlight_overrides = {
		all = {
			CmpBorder = { fg = "#3E4145" },
		},
		latte = {
			Normal = { fg = ucolors.darken(latte.base, 0.7, latte.mantle) },
		},
		frappe = {
			TSConstBuiltin = { fg = frappe.peach, style = {} },
			TSConstant = { fg = frappe.sky },
			TSComment = { fg = frappe.surface2, style = { "italic" } },
		},
		macchiato = {
			LineNr = { fg = macchiato.overlay1 }
		},
		mocha = {
			Comment = { fg = mocha.flamingo },
		},
	},
}
```

Additionally, if you want to load other custom highlights later, you may use this function:

```lua
require("catppuccin.lib.highlighter").syntax()
```

For example:

```lua
local colors = require("catppuccin.palettes").get_palette() -- fetch colors from palette
require("catppuccin.lib.highlighter").syntax({
	Comment = { fg = colors.surface0 }
})
```

> Note: custom highlights loaded using the `require("catppuccin.lib.highlighter").syntax()` function **won't** be pre-compiled.

#### Overwriting colors
Colors can be overwritten using `color_overrides` starting from [this line](https://github.com/ayamir/nvimdots/blob/94f153471068c5145b2036be1f688c0f954cd973/lua/modules/ui/config.lua#L268), like so:

```lua
require("catppuccin").setup {
	color_overrides = {
		all = {
			text = "#FFFFFF",
		},
		latte = {
			base = "#FF0000",
			mantle = "#242424",
			crust = "#474747",
		},
		frappe = {},
		macchiato = {},
		mocha = {},
	}
}
``` 

### Customizing auto-compile Hook
#### Available Compile Commands

```vim
:CatppuccinCompile " Create/update the compile file
:CatppuccinClean " Delete compiled file
:CatppuccinStatus " Compile status
```

Catppuccin also provides these functions to work with the catppuccin compiler:

```lua
local catppuccin = require('catppuccin')
-- Create/update the compile files
catppuccin.compile()
-- Delete compiled files
catppuccin.clean()
-- Show compile status
catppuccin.status()
```

#### Post-install/update hooks
- You may add `:CatppuccinCompile` to post-install/update hooks [here](https://github.com/ayamir/nvimdots/blob/94f153471068c5145b2036be1f688c0f954cd973/lua/modules/ui/plugins.lua#L7-L11), like so:
```lua
ui["catppuccin/nvim"] = {
	opt = false,
	as = "catppuccin",
	config = conf.catppuccin,
	run = ":CatppuccinCompile"
}
```

- If you want catppuccin setup function to actually reload without restarting nvim, set this option [here](https://github.com/ayamir/nvimdots/blob/94f153471068c5145b2036be1f688c0f954cd973/lua/core/pack.lua#L39-L61). Choose the appropriate place depending on your OS.
```lua
packer.init({
	auto_reload_compiled = true,
	.........
})
```

### Want to know more details?
- Visit catppuccin on [github](https://github.com/catppuccin/nvim)!
`````````

<br />
</details>

*****

# Guide on how to use the catppuccin colorscheme
## What is Catppuccin? <sup><a href="https://www.reddit.com/r/neovim/comments/r9619t/comment/hna3hz8">[1]</a></sup>
> Catppuccin is a community-driven soothing pastel theme that aims to be the middle ground between low and high-contrast themes, providing a warm color palette with 26 eye-candy colors that are bright enough to be visible during the day, yet pale enough to be easy on your eyes throughout the night.

## Basic Usage
Modify [these lines](https://github.com/ayamir/nvimdots/blob/94f153471068c5145b2036be1f688c0f954cd973/lua/modules/ui/config.lua#L191-L267). _(Note: This link might be slightly different from `HEAD`, but it can be used as a reference.)_ See detailed explanation of each option below.

### General
These settings are unrelated to any group and are globally independent.
-   `vim.g.catppuccin_flavour`: _(Can be any one of: `latte`, `frappe`, `macchiato`, or `mocha`)_ this is **mandatory**. You **must** set this value in order to make catppuccin work correctly. Note that `latte` is a light colorscheme, and the rest are dark schemes; The `mocha` palette is the only one that has been modified to make catppuccin look like the v0.1 one. Check out [this PR](https://github.com/ayamir/nvimdots/pull/163) for details.
-   `transparent_background`: _(Boolean)_ if true, disables setting the background color.
-   `term_colors`: _(Boolean)_ if true, sets terminal colors (a.k.a., `g:terminal_color_0`).

### Dim inactive
This setting manages the ability to dim **inactive** splits/windows/buffers.
-   `enabled`: _(Boolean)_ if true, dims the background color of inactive window or buffer or split.
-   `shade`: _(string)_ sets the shade to apply to the inactive split or window or buffer.
-   `percentage`: _(number from 0 to 1)_ percentage of the shade to apply to the inactive window, split or buffer.

### Styles
Handles the style of general highlight groups _(see `:h highlight-args` for detailed explanation)_:
-   `comments`: _(Table)_ changes the style of comments.
-   `functions`: _(Table)_ changes the style of functions  _(e.g., [`set_auto_compile`](https://github.com/ayamir/nvimdots/blob/94f153471068c5145b2036be1f688c0f954cd973/lua/modules/ui/config.lua#L188) in config)_.
-   `keywords`: _(Table)_ changes the style of keywords _(e.g., `local`)_.
-   `strings`: _(Table)_ changes the style of strings.
-   `variables`: _(Table)_ changes the style of variables.
-   `properties`: _(Table)_ changes the style of a phantom field with only getter and/or setter _(e.g., field access `tbl.field`)_.
-   `operators`: _(Table)_ changes the style of operators.
-   `conditionals`: _(Table)_ changes the style of conditional check keywords _(e.g., `if`)_.
-   `loops`: _(Table)_ changes the style of loop keywords _(e.g., `for`)_.
-   `booleans`: _(Table)_ changes the style of booleans.
-   `numbers`: _(Table)_ changes the style of numbers.
-   `types`: _(Table)_ changes the style of types _(e.g., `int`)_.

### Integrations
These integrations allow catppuccin to set the theme of various plugins. To enable an integration you need to set it to `true`.

### Using the auto-compile feature
> Catppuccin is a highly customizable and configurable colorscheme. This does however come at the cost of complexity and execution time.

Catppuccin can pre-compute the results of configuration and store the results in a compiled lua file. These pre-cached values are later used to set highlights. The cached file is stored at `vim.fn.stdpath("cache") .. "/catppuccin"` by default _(use `:lua print(vim.fn.stdpath("cache") .. "/catppuccin")` to see where it locates on your computer)_. You may change this behavior by modifying [this line](https://github.com/ayamir/nvimdots/blob/94f153471068c5145b2036be1f688c0f954cd973/lua/modules/ui/config.lua#L202).

If you would like to disable this feature, set [`enable_compile`](https://github.com/ayamir/nvimdots/blob/94f153471068c5145b2036be1f688c0f954cd973/lua/modules/ui/config.lua#L187) to false. However, this is **NOT** recommended because it could result in a slower startup time.

There's also an auto-compile hook that could compile catppuccin every time packer compiles _(i.e., the execution of `:PackerCompile`)_. Disable this behavior by commenting out [this line](https://github.com/ayamir/nvimdots/blob/94f153471068c5145b2036be1f688c0f954cd973/lua/modules/ui/config.lua#L188). _(You may also want to see its implementation - that's right above this line)_. However, after disabling this feature, you'll have to **manually** compile catppuccin using `:CatppuccinCompile` every time you modify its settings! _(If you enable catppuccin's compile feature, of course)._ Otherwise, the changes will **NOT** have any effect.

### Customizing the palette
Not satisfied with the current appearance? You may modify the palette yourself, like `mocha`!

#### Get catppuccin colors
```lua
local latte = require("catppuccin.palettes").get_palette "latte"
local frappe = require("catppuccin.palettes").get_palette "frappe"
local macchiato = require("catppuccin.palettes").get_palette "frappe"
local mocha = require("catppuccin.palettes").get_palette "mocha"
vim.g.catppuccin_flavour = "macchiato" -- Has to be set in order for empty argument to work
local colors = require("catppuccin.palettes").get_palette() -- g:catppuccin_flavour's palette
```

These lines would all return a table respectively, where the key is the name of the color and the value is its hex value.

#### Overwriting highlight groups
**Important Note:** If you also modified the color for any of the palettes at the same time, due to the fact that the changes made to the palette(s) when we're overwriting the highlight group have not been actually overwritten, and in the case of calling setup only once, the color we passed in has already been parsed (for example, `cp.pink -> "#F5C2E7"`), so it is **completely impossible** to obtain the changed palette, which means you need to manually set the new color after obtaining the palette, [like so](https://github.com/ayamir/nvimdots/blob/94f153471068c5145b2036be1f688c0f954cd973/lua/modules/ui/config.lua#L126-L164).

Global highlight groups can be overwritten starting from [this line](https://github.com/ayamir/nvimdots/blob/94f153471068c5145b2036be1f688c0f954cd973/lua/modules/ui/config.lua#L299) like so:

```lua
custom_highlights = {
	<hi_group> = { <fields> }
}
```

Here is an example:

```lua
vim.g.catppuccin_flavour = "macchiato"
local colors = require("catppuccin.palettes").get_palette() -- fetch colors from g:catppuccin_flavour palette
require("catppuccin").setup {
	custom_highlights = {
		Comment = { fg = colors.flamingo },
		TSConstBuiltin = { fg = colors.peach, style = {} },
		TSConstant = { fg = colors.sky },
		TSComment = { fg = colors.surface2, style = { "italic" } }
	}
}
```

Per flavour highlight groups can be overwritten like so:

```lua
highlight_overrides = {
	all = { -- Will be replaced with custom_highlights if it exists
		<hi_group> = { <fields> }
	}, -- Same for each flavour
	latte = {},
	frappe = {},
	macchiato = {},
	mocha = {},
}
```

Here is an example:

```lua
local ucolors = require "catppuccin.utils.colors"
local latte = require("catppuccin.palettes").get_palette "latte"
local frappe = require("catppuccin.palettes").get_palette "frappe"
local macchiato = require("catppuccin.palettes").get_palette "macchiato"
local mocha = require("catppuccin.palettes").get_palette "mocha"
vim.g.catppuccin_flavour = "macchiato"
local colors = require("catppuccin.palettes").get_palette() -- return vim.g.catppuccin_flavour palette
require("catppuccin").setup {
	highlight_overrides = {
		all = {
			CmpBorder = { fg = "#3E4145" },
		},
		latte = {
			Normal = { fg = ucolors.darken(latte.base, 0.7, latte.mantle) },
		},
		frappe = {
			TSConstBuiltin = { fg = frappe.peach, style = {} },
			TSConstant = { fg = frappe.sky },
			TSComment = { fg = frappe.surface2, style = { "italic" } },
		},
		macchiato = {
			LineNr = { fg = macchiato.overlay1 }
		},
		mocha = {
			Comment = { fg = mocha.flamingo },
		},
	},
}
```

Additionally, if you want to load other custom highlights later, you may use this function:

```lua
require("catppuccin.lib.highlighter").syntax()
```

For example:

```lua
local colors = require("catppuccin.palettes").get_palette() -- fetch colors from palette
require("catppuccin.lib.highlighter").syntax({
	Comment = { fg = colors.surface0 }
})
```

> Note: custom highlights loaded using the `require("catppuccin.lib.highlighter").syntax()` function **won't** be pre-compiled.

#### Overwriting colors
Colors can be overwritten using `color_overrides` starting from [this line](https://github.com/ayamir/nvimdots/blob/94f153471068c5145b2036be1f688c0f954cd973/lua/modules/ui/config.lua#L268), like so:

```lua
require("catppuccin").setup {
	color_overrides = {
		all = {
			text = "#FFFFFF",
		},
		latte = {
			base = "#FF0000",
			mantle = "#242424",
			crust = "#474747",
		},
		frappe = {},
		macchiato = {},
		mocha = {},
	}
}
``` 

#### Available Compile Commands

```vim
:CatppuccinCompile " Create/update the compile file
:CatppuccinClean " Delete compiled file
:CatppuccinStatus " Compile status
```

Catppuccin also provides these functions to work with the catppuccin compiler:

```lua
local catppuccin = require('catppuccin')
-- Create/update the compile files
catppuccin.compile()
-- Delete compiled files
catppuccin.clean()
-- Show compile status
catppuccin.status()
```

#### Post-install/update hooks
- You may add `:CatppuccinCompile` to post-install/update hooks [here](https://github.com/ayamir/nvimdots/blob/94f153471068c5145b2036be1f688c0f954cd973/lua/modules/ui/plugins.lua#L7-L11), like so:
```lua
ui["catppuccin/nvim"] = {
	opt = false,
	as = "catppuccin",
	config = conf.catppuccin,
	run = ":CatppuccinCompile"
}
```

- If you want catppuccin setup function to actually reload without restarting nvim, set this option [here](https://github.com/ayamir/nvimdots/blob/94f153471068c5145b2036be1f688c0f954cd973/lua/core/pack.lua#L39-L61). Choose the appropriate place depending on your OS.
```lua
packer.init({
	auto_reload_compiled = true,
	.........
})
```

### Want to know more details?
- Visit catppuccin on [github](https://github.com/catppuccin/nvim)!

**Comment by Jint-lzxy** [comment]
@ayamir Doc for catppuccin has been completed! Those changes can be picked and added to the wiki now 🥳
